plugins/lookup/random_string.py

```diff
@@ -0,0 +1,191 @@
+        type: str
+      base64:
+        description:
+        - Returns base64 encoded string.
+        type: bool
+"""
+
+EXAMPLES = r"""
+- name: Generate random string
+  ansible.builtin.debug:
+    var: lookup('community.general.random_string')
+  # Example result: 'DeadBeeF'
+
+- name: Generate random string with length 12
+  ansible.builtin.debug:
+    var: lookup('community.general.random_string', length=12)
+  # Example result: 'Uan0hUiX5kVG'
+
+- name: Generate base64 encoded random string
+  ansible.builtin.debug:
+    var: lookup('community.general.random_string', base64=True)
+  # Example result: 'NHZ6eWN5Qk0='
+
+- name: Generate a random string with 1 lower, 1 upper, 1 number and 1 special char (atleast)
+  ansible.builtin.debug:
+    var: lookup('community.general.random_string', min_lower=1, min_upper=1, min_special=1, min_numeric=1)
+  # Example result: '&Qw2|E[-'
+
+- name: Generate a random string with all lower case characters
+  debug:
+    var: query('community.general.random_string', upper=false, numbers=false, special=false)
+  # Example result: 'exolxzyz'
+
+- name: Generate random hexadecimal string
+  debug:
+    var: query('community.general.random_string', upper=false, lower=false, override_special=hex_chars, numbers=false)
+  vars:
+    hex_chars: '0123456789ABCDEF'
+  # Example result: 'D2A40737'
+"""
+
+RETURN = r"""
+  _raw:
+    description: A one-element list containing a random string
+    type: list
+    elements: str
+"""
+
+import base64
+import string
+import random
+
+from ansible.errors import AnsibleLookupError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils._text import to_bytes, to_text
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables=None, **kwargs):
+        number_chars = string.digits
+        lower_chars = string.ascii_lowercase
+        upper_chars = string.ascii_uppercase
+        special_chars = string.punctuation
+
+        self.set_options(var_options=variables, direct=kwargs)
+        length = self.get_option("length")
+        upper = self.get_option("upper")
+        lower = self.get_option("lower")
+        numbers = self.get_option("numbers")
+        special = self.get_option("special")
+        base64_flag = self.get_option("base64")
+        override_special = self.get_option("override_special")
+
+        if override_special:
+            special_chars = override_special
+
+        available_chars_set = ""
+        if upper:
+            available_chars_set += upper_chars
+        if lower:
+            available_chars_set += lower_chars
+        if numbers:
+            available_chars_set += number_chars
+        if special:
+            available_chars_set += special_chars
+
+        mapping = {
+            "min_numeric": number_chars,
+            "min_lower": lower_chars,
+            "min_upper": upper_chars,
+            "min_special": special_chars,
+        }
+
+        def get_random(chars, length):
```

Comment:
How about moving these functions either to class methods, or to global functions?

Reply:
OK.